.github/workflows/ci.yml

```diff
@@ -19,7 +19,6 @@ jobs:
         uses: actions-rs/toolchain@v1
```

Comment:
Are both this toolchain step and newly added one in line 50 necessary?

Reply:
I used those two steps to choose whether to input the toolchain (use the specified version or use the pinned version) depending on whether it is stable or not.
There may be a better way to do this that I just don't know about...... but it was removed by  a1fedce anyway.